src/Microsoft.AspNet.Mvc.ModelBinding/Internal/DictionaryHelper.cs

```diff
@@ -28,6 +29,16 @@ public static class DictionaryHelper
                     continue;
                 }
 
+                if (key.StartsWith("[", StringComparison.OrdinalIgnoreCase))
```

Comment:
Which means that the tests are not sufficient. Please add the missing tests for validation state as well.